src/app/custom-editor/custom-editor.component.ts

```diff
@@ -187,4 +214,18 @@ export class CustomEditorComponent implements OnInit, OnDestroy, OnChanges {
   getValue(): string {
     return this.editorContent ?? '';
   }
+
+  setAvailableViews(): void {
+    if (!this.isJson() && !this.isXml()) {
+      this.availableViews = editorViewsConst.filter((view: EditorView) => view == 'raw');
+      return;
+    }
+    if (!this.isXml()) {
+      this.availableViews = editorViewsConst.filter((view: EditorView) => view !== 'xml');
+      return;
+    }
+    if (!this.isJson()) {
+      this.availableViews = editorViewsConst.filter((view: EditorView) => view !== 'json');
+    }
+  }
```

Comment:
This feels like it could be done differently, why is `isXml` & `isJson` a seperate thing instead of a `getContentType` that returns `'json' | 'xml' | 'raw'`

Reply:
Fixed